tests/capture_tools_output/test_tools_output.py

```diff
@@ -18,14 +20,117 @@
 
 CLANG_VERSION = os.getenv("CLANG_VERSION", "12")
 
+TEST_REPO_COMMIT_PAIRS: List[Dict[str, str]] = [
+    dict(
+        repo="chocolate-doom/chocolate-doom",
+        commit="67715d6e2725322e6132e9ff99b9a2a3f3b10c83",
+    ),
+    dict(
+        repo="chocolate-doom/chocolate-doom",
+        commit="71091562db5b0e7853d08ffa2f110af49cc3bc0d",
+    ),
+    dict(
+        repo="libvips/libvips",
+        commit="fe82be345a5b654a76835a7aea5a804bd9ebff0a",
+    ),
+    dict(
+        repo="shenxianpeng/test-repo",
+        commit="662ad4cf90084063ea9c089b8de4aff0b8959d0e",
+    ),
+]
 
+
+def _translate_lines_changed_only_value(value: int) -> str:
+    """generates an id for tests that use lines-changed-only settings."""
+    ret_vals = ["all lines", "only added", "only diff"]
+    return ret_vals[value]
+
+
+def flush_prior_artifacts():
+    """flush output from any previous tests"""
+    cpp_linter.Globals.OUTPUT = ""
+    cpp_linter.Globals.FILES.clear()
+    cpp_linter.GlobalParser.format_advice.clear()
+    cpp_linter.GlobalParser.tidy_advice.clear()
+    cpp_linter.GlobalParser.tidy_notes.clear()
+
+
+def prep_repo(
+    monkeypatch: pytest.MonkeyPatch,
+    repo: str,
+    commit: str,
+):
+    """Setup a test repo to run the rest of the tests in this module."""
+    for name, value in zip(["GITHUB_REPOSITORY", "GITHUB_SHA"], [repo, commit]):
+        monkeypatch.setattr(cpp_linter.run, name, value)
+
+    flush_prior_artifacts()
+    test_diff = Path(__file__).parent / repo / f"{commit}.diff"
+    monkeypatch.setattr(
+        cpp_linter.Globals, "FILES", parse_diff(test_diff.read_text(encoding="utf-8"))
+    )
+
+
+def prep_tmp_dir(
+    tmp_path: Path,
+    monkeypatch: pytest.MonkeyPatch,
+    repo: str,
+    commit: str,
+    copy_configs: bool = False,
+    lines_changed_only: int = 0,
+):
+    """Some extra setup for test's temp directory to ensure needed files exist."""
+    monkeypatch.chdir(str(tmp_path))
+    prep_repo(
+        monkeypatch,
+        repo=repo,
+        commit=commit,
+    )
+    if copy_configs:
+        for config in ("format", "tidy"):
+            shutil.copyfile(
+                str(Path(__file__).parent / repo / f".clang-{config}"),
+                str(tmp_path / f".clang-{config}"),
+            )
+    # Make a folder to download the needed files in the tests' temp folder. This is
+    # meant to avoid re-downloading the same files for multiple tests run against the
+    # same sample repo.
+    repo_cache = tmp_path.parent / repo
+    repo_cache.mkdir(parents=True, exist_ok=True)
+    monkeypatch.chdir(str(repo_cache))
+    filter_out_non_source_files(["c", "h"], [".github"], [], lines_changed_only)
+    cpp_linter.run.verify_files_are_present()
+    repo_path = tmp_path / repo.split("/")[1]
+    shutil.copytree(
+        str(repo_cache),
+        str(repo_path),
+        ignore=shutil.ignore_patterns("\\.changed_files.json"),
+    )
```

Comment:
This doesn't correctly compensate for testing multiple commits of the same repo. `verify_files_are_present()` has no guarantee that the existing file's contents are up-to-date. This may be lead to using outdated cache files if both commits' diffs had changes to the same file.

Reply:
fixed with
```diff
-    repo_cache = tmp_path.parent / repo
+    repo_cache = tmp_path.parent / repo / commit
```